bundle/regal/rules/bugs/deprecated_builtin.rego

```diff
@@ -0,0 +1,39 @@
+# METADATA
+# description: Avoid using deprecated built-in functions
+package regal.rules.bugs["deprecated-builtin"]
+
+import rego.v1
+
+import data.regal.ast
+import data.regal.config
+import data.regal.result
+
+report contains violation if {
+	deprecated_builtins := {
+		"any", "all", "re_match", "net.cidr_overlap", "set_diff", "cast_array",
+		"cast_set", "cast_string", "cast_boolean", "cast_null", "cast_object",
+	}
+
+	# if none of the deprecated built-ins are in the
+	# capabilities for the target, bail out early
+	any_deprecated_builtin(object.keys(config.capabilities.builtins), deprecated_builtins)
+
+	some ref in ast.all_refs
+
+	ref[0].value[0].type == "var"
+	not ref[0].value[0].value in {"input", "data"}
```

Comment:
Correct, but redundant, isn't it? This could never pass `name in deprecated_builtins` since that set contains nothing starting with input or data.

Reply:
Ah, yes, copy-pasta 🍝  I'll fix

Reply:
But OTOH, it'll avoid having to concat the parts in the first place on long `input`/`data` refs? So I think this is fine to keep.